pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -71,3 +85,84 @@ func (p RESTStorageProvider) v1alpha1Storage(apiResourceConfigSource serverstora
 func (p RESTStorageProvider) GroupName() string {
 	return flowcontrol.GroupName
 }
+
+func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
+	return PostStartHookName, func(hookContext genericapiserver.PostStartHookContext) error {
+		flowcontrolClientSet := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+		// ensure the pre-defined objects if the exempt flow-schema is present.
+		shouldEnsure, err := shouldEnsureAllPredefined(flowcontrolClientSet)
+		if err != nil {
+			return err
+		}
+		if shouldEnsure {
+			return ensureAllPredefined(
+				flowcontrolClientSet,
+				flowcontrolbootstrap.PredefinedFlowSchemas(),
+				flowcontrolbootstrap.PredefinedPriorityLevelConfigurations(),
+			)
+		}
+		return nil
+	}, nil
+}
+
+func shouldEnsureAllPredefined(flowcontrolClientSet flowcontrolclient.FlowcontrolV1alpha1Interface) (bool, error) {
+	if _, err := flowcontrolClientSet.FlowSchemas().Get(flowcontrol.PriorityLevelConfigurationNameExempt, metav1.GetOptions{}); err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func ensureAllPredefined(flowcontrolClientSet flowcontrolclient.FlowcontrolV1alpha1Interface, flowSchemas []*flowcontrolv1alpha1.FlowSchema, priorityLevels []*flowcontrolv1alpha1.PriorityLevelConfiguration) error {
+	err := wait.PollImmediate(time.Second, 30*time.Second, func() (done bool, err error) {
+		for _, flowSchema := range flowSchemas {
+			_, err := flowcontrolClientSet.FlowSchemas().Get(flowSchema.Name, metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					// TODO(yue9944882): use server-side apply to ensure the defaults
```

Comment:
I do not think we need server-side apply.  We only need something very simple here: create the object if it does not already exist.  We do not even need to do the Get first.  Just try the Create.

Reply:
i remember we agreed to use server-side apply to create the objects in a past meeting.. can't recall the detail..

Reply:
Given that we're now segregating the objects into immutable and suggested ones, I don't think SSA is needed any longer.